src/Foundation/Site.php

```diff
@@ -11,281 +11,53 @@
         }
 
         date_default_timezone_set('UTC');
 
-        $app = new Application($this->basePath, $this->publicPath);
-
-        if ($this->storagePath) {
-            $app->useStoragePath($this->storagePath);
-        }
-
-        $app->instance('env', 'production');
-        $app->instance('flarum.config', $this->getConfig());
-        $app->instance('config', $config = $this->getIlluminateConfig($app));
-
-        $this->registerLogger($app);
-
-        $this->registerCache($app);
-
-        $app->register(DatabaseServiceProvider::class);
-        $app->register(MigrationServiceProvider::class);
-        $app->register(SettingsServiceProvider::class);
-        $app->register(LocaleServiceProvider::class);
-        $app->register(BusServiceProvider::class);
-        $app->register(FilesystemServiceProvider::class);
-        $app->register(HashServiceProvider::class);
-        $app->register(MailServiceProvider::class);
-        $app->register(ViewServiceProvider::class);
-        $app->register(ValidationServiceProvider::class);
-
-        $app->register(BusProvider::class);
-
-        if ($app->isInstalled() && $app->isUpToDate()) {
-            $settings = $app->make(SettingsRepositoryInterface::class);
-
-            $config->set('mail.driver', $settings->get('mail_driver'));
-            $config->set('mail.host', $settings->get('mail_host'));
-            $config->set('mail.port', $settings->get('mail_port'));
-            $config->set('mail.from.address', $settings->get('mail_from'));
-            $config->set('mail.from.name', $settings->get('forum_title'));
-            $config->set('mail.encryption', $settings->get('mail_encryption'));
-            $config->set('mail.username', $settings->get('mail_username'));
-            $config->set('mail.password', $settings->get('mail_password'));
-
-            $app->register(DiscussionServiceProvider::class);
-            $app->register(FormatterServiceProvider::class);
-            $app->register(FrontendServiceProvider::class);
-            $app->register(GroupServiceProvider::class);
-            $app->register(NotificationServiceProvider::class);
-            $app->register(PostServiceProvider::class);
-            $app->register(SearchServiceProvider::class);
-            $app->register(UserServiceProvider::class);
-
-            $app->register(ApiServiceProvider::class);
-            $app->register(ForumServiceProvider::class);
-            $app->register(AdminServiceProvider::class);
-
-            foreach ($this->extenders as $extender) {
-                // TODO: Create extenders architecture
-                // $extender->apply($app);
-            }
-
-            $app->register(ExtensionServiceProvider::class);
+        if (static::hasConfigFile($paths['base'])) {
+            return new InstalledSite(
+                $paths['base'],
+                $paths['public'],
+                static::loadConfig($paths['base'])
+            );
+        } else {
```

Comment:
`else` not required here

Reply:
True, I am a big fan of early returns as well. In this particular case, I did it on purpose, though, because both options had the same importance to me. (I typically use early returns for separating exceptional cases from the happy path, which can apparently be subjective.)

Is that fine or too inconsistent? :wink:

Reply:
Fair reasoning!